docs/pages/implementation/nuts-auth.rst

```diff
@@ -0,0 +1,225 @@
+    }
+
+This bearer token can be used to get an access token at the custodian side.
+
+Access token
+************
+
+The access token is issued by the authorization server at the custodian side. According to spec it may be anything as long as it's BASE64 encoded and it can fit in a HTTP header. The authorization server embedded in the Nuts OS implementation uses JWTs as access tokens.
+
+To get an access token, perform the following steps:
+
+- find the authorization server endpoint
+- retrieve the access token
+- use the access token
+
+One way of obtaining the right endpoint is to use the Nuts registry. The Nuts registry holds OAuth endpoints for each vendor, care organization and service combination.
+
+.. note::
+
+    This is work in progress. The API hasn't been defined yet on the registry.
+
+We'll use the access token endpoint on the Nuts OS implementation as reference endpoint:
+
+.. code-block::
+
+    POST /auth/accesstoken HTTP/1.1
+    Host: server.example.com
+    Content-Type: application/x-www-form-urlencoded
+
+    grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Ajwt-bearer
+    &assertion=eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCIsIng1YyI...WSOR9zuNdW_nYHwTDQI4z7pVkUqgdEd-BhA
+
+the ``grant-type`` is fixed and equals **urn:ietf:params:oauth:grant-type:jwt-bearer**. The ``assertion`` is the **bearer token** from the previous paragraph. This request must be send over TLS and the sender must use a **client certificate**. `RFC003 <https://nuts-foundation.gitbook.io/drafts/rfc/rfc003-oauth2-authorization>`_ and `RFC008 <https://nuts-foundation.gitbook.io/drafts/rfc/rfc008-certificate-structure>`_ specify the requirements for the client certificate. The Nuts OS implementation contains a convenient endpoint for getting a client certificate signed by the vendor CA:
+
+.. code-block::
+
+    POST /crypto/certificate/tls HTTP/1.1
+    Host: server.example.com
+    Content-Type: application/octet-stream
+
+    -----BEGIN PUBLIC KEY-----
+    MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAExr9OdS3meIrzaqNMi+RWIT5v4hcA
+    AJwvfhf8Wy+jh97W8fPGwVZvUcRTrSESvGBV+kas4pd+K4E335Ifd1+XIg==
+    -----END PUBLIC KEY-----
+
+The result will be the client certificate in pem encoding. The client is responsible for the storage of the key material. If the authorization server accepts the bearer token it'll return an access token:
```

Comment:
PEM

Reply:
> The client is responsible for the storage of the key material

Since it's managed by the Nuts node the "client" sounds strange (like we mean the CLI client or HTTP client). Maybe just warn that key material produced by the Nuts node should be managed/backuped/etc?

Reply:
Well, it's not the node that stores the key. It's the app that makes requests to the Nuts node.